### ISSUE TITLE
Nearby search not restored properly from bundle

### ISSUE THREAD

**Issue by SammysHP** [cgeo/cgeo#11911]
### Describe your problem!

On a low memory situation when the app was in the background and the activity was restored from the bundle it contains only results that are stored in the database (e.g. if they were opened previously).

### How to reproduce?

1. Open nearby search
2. Switch to another app that forces c:geo to be unloaded.
3. Return to c:geo

### Actual result after these steps?

Not all caches of the nearby search result are restored.

### Expected result after these steps?

All caches of the search result are restored.

### Reproducible

Yes

### c:geo Version

2021.10.15-NB-fd3f98e

### System information

```text
## System information

c:geo version: 2021.10.15-NB-fd3f98e

Device:
-------
- Device type: Redmi Note 4 (lineage_mido, Xiaomi)
- Available processors: 8
- Android version: 7.1.2
- Android build: lineage_mido-userdebug 7.1.2 NJH47F 57ca1c7c3e
- Screen resolution: 1080x1920px (400x687dp)
- Pixel density: 2.7
- System font scale: 1.0 / used scale: 1.0
- Sailfish OS detected: false
- Google Play services: disabled - 21.39.16 (040408-402663742)

Sensor and location:
-------
- Low power mode: inactive
- Compass capabilities: yes
- Rotation vector sensor: present
- Orientation sensor: present
- Magnetometer & Accelerometer sensor: present
- Direction sensor used: rotation vector

Program settings:
-------
- Settings: v8, Count:184
- Set language: de_DE (system default)
- System date format: dd.MM.yy
- Time zone: GMT+02:00
- Debug mode active: no
- Last backup: never
- Routing mode: Straight
- Live map mode: true
- OSM multi-threading: true / threads: 4
- Map: Freizeitkarte_NIEDERSACHSEN (Offline)
  - Id: cgeo.geocaching.maps.mapsforge.MapsforgeMapProvider$OfflineMapSource:primary:Locus/mapsVector/Freizeitkarte_NIEDERSACHSEN.map
  - Atts: (c) Map: FZK project (free for private use); Map data: OpenStreetMap contributors; Contour data: U.S. Geological Survey and J. de Ferranti
  - Theme: fzk-outdoor-contrast-v5.zip:fzk-outdoor-contrast-v5/fzk-outdoor-contrast-v5.xml

Filters:
-------
- Hide waypoints: -
- LIVE: Schwierigkeit / Gelände:1.0-4.5 / 1.0-4.5,Status:Gefunden=Nein,Eigener=Nein ([:advanced=false:inconclusive=false]AND(type;difficulty_terrain:d=1.0:d=4.5:t=1.0:t=4.5;status:owned_no:found_no))
- OFFLINE: Keine ([:advanced=false:inconclusive=false]AND(type;difficulty_terrain:d=-:d=-:t=-:t=-;status))

Services:
-------
- Geocaching sites enabled:
   geocaching.com: Logged in (Anmeldung OK) / PREMIUM
   opencaching.de: Logged in (Anmeldung OK)
- Geocaching.com date format: MM/dd/yyyy
- Routing: external / BRouter installed: true
- Installed c:geo plugins:  none

Permissions & paths:
-------
- Fine location permission: granted
- Write external storage permission: granted
- System internal c:geo dir: /data/user/0/cgeo.geocaching (2,5 GB free) v2 internal isDir(10 entries)
- Legacy User storage c:geo dir: /storage/emulated/0/cgeo (2,5 GB free) v2 external non-removable isDir(8 entries)
- Geocache data: /storage/emulated/0/Android/data/cgeo.geocaching/files/GeocacheData (2,5 GB free) v2 external non-removable isDir(38 entries)
- Internal theme sync (is turned ON): /data/user/0/cgeo.geocaching/MapThemeData (2,5 GB free) v2 internal isDir(4 entries)
- Public Folders: #11
  - BASE: /cgeo (User-Defined)[/cgeo[DOCUMENT#0:p-content://com.android.externalstorage.documents/tree/primary%3Acgeo::]] (Uri: content://com.android.externalstorage.documents/tree/primary%3Acgeo/document/primary%3Acgeo, Av:true, files:>=5, dirs:>=8, totalFileSize:>=2,0 MB, free space: 2,5 GB, files on device: 1563648)
  - OFFLINE_MAPS: /Locus/mapsVector (User-Defined)[/Locus/mapsVector[DOCUMENT#0:p-content://com.android.externalstorage.documents/tree/primary%3ALocus%2FmapsVector::]] (Uri: content://com.android.externalstorage.documents/tree/primary%3ALocus%2FmapsVector/document/primary%3ALocus%2FmapsVector, Av:true, files:>=102, dirs:>=15, totalFileSize:>=289,7 MB, free space: 2,5 GB, files on device: 1563648)
  - OFFLINE_MAP_THEMES: /cgeo/themes (User-Defined)[/cgeo/themes[DOCUMENT#0:p-content://com.android.externalstorage.documents/tree/primary%3Acgeo%2Fthemes::]] (Uri: content://com.android.externalstorage.documents/tree/primary%3Acgeo%2Fthemes/document/primary%3Acgeo%2Fthemes, Av:true, files:3, dirs:0, totalFileSize:1,0 MB, free space: 2,5 GB, files on device: 1563648)
  - LOGFILES: /cgeo/logfiles (Default)[/cgeo/logfiles[PERSISTABLE_FOLDER(BASE)#1:p-content://com.android.externalstorage.documents/tree/primary%3Acgeo::/logfiles]] (Uri: content://com.android.externalstorage.documents/tree/primary%3Acgeo/document/primary%3Acgeo%2Flogfiles, Av:true, files:0, dirs:0, totalFileSize:0 B, free space: 2,5 GB, files on device: 1563648)
  - GPX: /cgeo/gpx (Default)[/cgeo/gpx[PERSISTABLE_FOLDER(BASE)#1:p-content://com.android.externalstorage.documents/tree/primary%3Acgeo::/gpx]] (Uri: content://com.android.externalstorage.documents/tree/primary%3Acgeo/document/primary%3Acgeo%2Fgpx, Av:true, files:1, dirs:0, totalFileSize:996,5 KB, free space: 2,5 GB, files on device: 1563648)
  - BACKUP: /cgeo/backup (Default)[/cgeo/backup[PERSISTABLE_FOLDER(BASE)#1:p-content://com.android.externalstorage.documents/tree/primary%3Acgeo::/backup]] (Uri: content://com.android.externalstorage.documents/tree/primary%3Acgeo/document/primary%3Acgeo%2Fbackup, Av:true, files:0, dirs:0, totalFileSize:0 B, free space: 2,5 GB, files on device: 1563648)
  - FIELD_NOTES: /cgeo/field-notes (Default)[/cgeo/field-notes[PERSISTABLE_FOLDER(BASE)#1:p-content://com.android.externalstorage.documents/tree/primary%3Acgeo::/field-notes]] (Uri: content://com.android.externalstorage.documents/tree/primary%3Acgeo/document/primary%3Acgeo%2Ffield-notes, Av:true, files:1, dirs:0, totalFileSize:206 B, free space: 2,5 GB, files on device: 1563648)
  - SPOILER_IMAGES: /cgeo/GeocachePhotos (Default)[/cgeo/GeocachePhotos[PERSISTABLE_FOLDER(BASE)#1:p-content://com.android.externalstorage.documents/tree/primary%3Acgeo::/GeocachePhotos]] (Uri: content://com.android.externalstorage.documents/tree/primary%3Acgeo/document/primary%3Acgeo%2FGeocachePhotos, Av:true, files:0, dirs:0, totalFileSize:0 B, free space: 2,5 GB, files on device: 1563648)
  - ROUTING_BASE: /cgeo/routing (Default)[/cgeo/routing[PERSISTABLE_FOLDER(BASE)#1:p-content://com.android.externalstorage.documents/tree/primary%3Acgeo::/routing]] (Uri: content://com.android.externalstorage.documents/tree/primary%3Acgeo/document/primary%3Acgeo%2Frouting, Av:true, files:7, dirs:1, totalFileSize:81,2 KB, free space: 2,5 GB, files on device: 1563648)
  - ROUTING_TILES: /cgeo/routing/segments4 (Default)[/cgeo/routing/segments4[PERSISTABLE_FOLDER(ROUTING_BASE)#1:p-content://com.android.externalstorage.documents/tree/primary%3Acgeo::/routing/segments4]] (Uri: content://com.android.externalstorage.documents/tree/primary%3Acgeo/document/primary%3Acgeo%2Frouting%2Fsegments4, Av:true, files:0, dirs:0, totalFileSize:0 B, free space: 2,5 GB, files on device: 1563648)
  - TEST_FOLDER: [Legacy]/data/user/0/cgeo.geocaching/files/unittest (Default)[/data/user/0/cgeo.geocaching/files/unittest[FILE#1:p-file:///data/user/0/cgeo.geocaching/files::/unittest]] (Uri: file:///data/user/0/cgeo.geocaching/files/unittest, Av:true, files:0, dirs:0, totalFileSize:0 B, free space: 2,5 GB, files on device: -1)
- Map render theme path: fzk-outdoor-contrast-v5.zip:fzk-outdoor-contrast-v5/fzk-outdoor-contrast-v5.xml
- PersistedDocumentUris: #1
- TRACK: null
- Persisted Uri Permissions: #3
  - content://com.android.externalstorage.documents/tree/primary%3ALocus%2FmapsVector (20. Feb., 09:09):RW
  - content://com.android.externalstorage.documents/tree/primary%3Acgeo (20. Feb., 09:08):RW
  - content://com.android.externalstorage.documents/tree/primary%3Acgeo%2Fthemes (9. Mai, 07:50):RW
- Database: /data/user/0/cgeo.geocaching/databases/data (v98, Size:1,0 MB) on system internal storage

--- End of system information ---
```

**Comment by SammysHP** [comment]
On another try I got a result from Paris, over 600 km away, when returning to c:geo.

**Comment by fm-sys** [comment]
> Probably fixed with PR #11918

let's close this issue as it was most likely fixed...
